### PR TITLE
Add latest alert to public api alert groups endpoint

### PR DESCRIPTION
**Pull request by iskhakov**
# What this PR does

Added last alert information 
and optimized the API call so it makes 10x less queries by:
* prefetching chatops messages (based on @vadimkerr 's https://github.com/grafana/oncall/pull/4738)
* using `enrich` from private api

Previously: 
Screenshot 2024-09-24 at 4 47 00 PM
Now:
Screenshot 2024-09-24 at 4 44 56 PM

## Which issue(s) this PR closes

Related to [issue link here]

## Checklist

- [ ] Unit, integration, and e2e (if applicable) tests updated
- [ ] Documentation added (or `pr:no public docs` PR label added if not required)
- [ ] Added the relevant release notes label (see labels prefixed w/ `release:`). These labels dictate how your PR will
    show up in the autogenerated release notes.

### CODE REVIEW

**Comment by iskhakov** [comment]
we need to copy enrich() from private api for better performance

File: engine/apps/public_api/serializers/alert_groups.py [hunk 1]
```diff
@@ -18,9 +22,24 @@ class AlertGroupSerializer(EagerLoadingMixin, serializers.ModelSerializer):
...
+            to_attr="prefetched_slack_messages",
+        ),
+        Prefetch(
+            "telegram_messages",
+            queryset=TelegramMessage.objects.filter(
+                chat_id__startswith="-", message_type=TelegramMessage.ALERT_GROUP_MESSAGE
+            ).order_by("id")[:1],
+            to_attr="prefetched_telegram_messages",
+        ),
+    ]
```
Comment (vadimkerr) [inline engine/apps/public_api/serializers/alert_groups.py:42]: nit:

```suggestion
    PREFETCH_RELATED = [
        "labels",
        Prefetch(
            "slack_messages",
            queryset=SlackMessage.objects.select_related("_slack_team_identity").order_by("created_at")[:1],
            to_attr="prefetched_slack_messages",
        ),
        Prefetch(
            "telegram_messages",
            queryset=TelegramMessage.objects.filter(
                chat_id__startswith="-", message_type=TelegramMessage.ALERT_GROUP_MESSAGE
            ).order_by("id")[:1],
            to_attr="prefetched_telegram_messages",
        ),
    ]
```

File: engine/apps/public_api/serializers/alert_groups.py [hunk 2]
```diff
@@ -56,3 +73,19 @@ def get_route_id(self, obj):
             return obj.channel_filter.public_primary_key
         else:
             return None
+
+    def get_last_alert(self, obj):
+        if hasattr(obj, "last_alert") and obj.last_alert:
+            return AlertSerializer(obj.last_alert).data
+        # Fall back to the latest alert in the group if the last_alert is not set by AlertGroupEnrichingMixin
+        if obj.alerts.exists():
+            return AlertSerializer(obj.alerts.latest("created_at")).data
```
Comment (vadimkerr) [inline engine/apps/public_api/serializers/alert_groups.py:82]: maybe simplify this to 1 sql query?

File: engine/apps/public_api/serializers/alert_groups.py [hunk 3]
```diff
@@ -56,3 +73,19 @@ def get_route_id(self, obj):
...
+        return None
+
+    def get_alerts_count(self, obj):
+        if hasattr(obj, "alerts_count") and obj.alerts_count is not None:
```
Comment (vadimkerr) [inline engine/apps/public_api/serializers/alert_groups.py:86]: I _think_ obj.alerts_count can't be None

File: engine/apps/public_api/serializers/alert_groups.py [hunk 4]
```diff
@@ -56,3 +73,19 @@ def get_route_id(self, obj):
...
+            return obj.alerts_count
+        # Fall back to the count of alerts in the group if the alerts_count is not set by AlertGroupEnrichingMixin
+        if obj.alerts.exists():
+            return obj.alerts.count()
+        return 0
```
Comment (vadimkerr) [inline engine/apps/public_api/serializers/alert_groups.py:91]:
```suggestion
        return obj.alerts.count()
```

File: engine/apps/api/serializers/alert_group.py [hunk 5]
```diff
@@ -138,21 +137,20 @@ class AlertGroupListSerializer(
...
-            ),
-            Prefetch(
-                "telegram_messages",
-                queryset=TelegramMessage.objects.filter(
-                    chat_id__startswith="-", message_type=TelegramMessage.ALERT_GROUP_MESSAGE
-                ).order_by("id")[:1],
-                to_attr="prefetched_telegram_messages",
-            ),
-        ]
+    PREFETCH_RELATED += [
```
Comment (vadimkerr) [inline engine/apps/api/serializers/alert_group.py:140]: nit: we probably don't need += after removing ALERT_GROUP_LIST_TRY_PREFETCH

File: engine/settings/base.py [hunk 6]
```diff
@@ -42,7 +42,7 @@
 MIRAGE_CIPHER_MODE = "CBC"
 
 # SECURITY WARNING: don't run with debug turned on in production!
-DEBUG = False
+DEBUG = getenv_boolean("DEBUG", default=False)
```
Comment (vadimkerr) [inline engine/settings/base.py:45]: nit: things work on my machine without this change. [settings/dev.py](https://github.com/grafana/oncall/blob/6511e439c56b5c3d9f3d82811bd04abe57ca1da5/engine/settings/dev.py#L9)

**Review by vadimkerr — APPROVED** [review]
lgtm, left some nit comments

**Review by alyssawada — APPROVED** [review]

**Pull request merged**